pkg/resources/provisioning.cattle.io/v1/cluster/validator_test.go

```diff
@@ -200,3 +205,373 @@ func TestValidateMachinePoolName(t *testing.T) {
 		})
 	}
 }
+
+func Test_validateAgentDeploymentCustomization(t *testing.T) {
+	type args struct {
+		customization *v1.AgentDeploymentCustomization
+		path          *field.Path
+	}
+	type validation func(t *testing.T, err field.ErrorList)
+
+	validateFailedPaths := func(s []string) validation {
+		return func(t *testing.T, err field.ErrorList) {
+			errPaths := make([]string, len(err), len(err))
```

Comment:
specify helper functions for better stack trace on error.
```suggestion
            t.Helper()
			errPaths := make([]string, len(err), len(err))
```